modules/base/rendering/pointcloud/renderablepolygoncloud.cpp

```diff
@@ -0,0 +1,194 @@
+    createPolygonTexture();
+}
+
+void RenderablePolygonCloud::deinitializeGL() {
+    glDeleteBuffers(1, &_polygonVbo);
+    _polygonVbo = 0;
+    glDeleteVertexArrays(1, &_polygonVao);
+    _polygonVao = 0;
+
+    glDeleteTextures(1, &_pTexture);
+}
+
+void RenderablePolygonCloud::bindTextureForRendering() const {
+    glBindTexture(GL_TEXTURE_2D, _pTexture);
+}
+
+void RenderablePolygonCloud::createPolygonTexture() {
+    ZoneScoped;
+
+    LDEBUG("Creating Polygon Texture");
+    constexpr gl::GLsizei TexSize = 512;
+
+    glGenTextures(1, &_pTexture);
+    glBindTexture(GL_TEXTURE_2D, _pTexture);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+    // Stopped using a buffer object for GL_PIXEL_UNPACK_BUFFER
+    glBindBuffer(GL_PIXEL_UNPACK_BUFFER, 0);
+    glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA8, TexSize, TexSize, 0, GL_RGBA, GL_BYTE, nullptr);
+
+    renderToTexture(_pTexture, TexSize, TexSize);
+}
+
+void RenderablePolygonCloud::renderToTexture(GLuint textureToRenderTo,
+                                             GLuint textureWidth, GLuint textureHeight)
+{
+    LDEBUG("Rendering to Texture");
+
+    // Saves initial Application's OpenGL State
+    GLint defaultFBO;
+    GLint viewport[4];
+    glGetIntegerv(GL_FRAMEBUFFER_BINDING, &defaultFBO);
+    glGetIntegerv(GL_VIEWPORT, viewport);
+
+    GLuint textureFBO;
+    glGenFramebuffers(1, &textureFBO);
+    glBindFramebuffer(GL_FRAMEBUFFER, textureFBO);
+    GLenum drawBuffers[1] = { GL_COLOR_ATTACHMENT0 };
+    glDrawBuffers(1, drawBuffers);
+
+    glFramebufferTexture(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, textureToRenderTo, 0);
+
+    glViewport(viewport[0], viewport[1], textureWidth, textureHeight);
+
+    loadPolygonGeometryForRendering();
+    renderPolygonGeometry(_polygonVao);
+
+    // Restores Applications' OpenGL State
+    glBindFramebuffer(GL_FRAMEBUFFER, defaultFBO);
+    glViewport(viewport[0], viewport[1], viewport[2], viewport[3]);
+
+    glDeleteBuffers(1, &_polygonVbo);
+    glDeleteVertexArrays(1, &_polygonVao);
+    glDeleteFramebuffers(1, &textureFBO);
+}
+
+void RenderablePolygonCloud::loadPolygonGeometryForRendering() {
+    if (_polygonVao == 0) {
+        glGenVertexArrays(1, &_polygonVao);
+    }
+    if (_polygonVbo == 0) {
+        glGenBuffers(1, &_polygonVbo);
+    }
+
+    glBindVertexArray(_polygonVao);
+    glBindBuffer(GL_ARRAY_BUFFER, _polygonVbo);
+
+    constexpr std::array<GLfloat, 4> VertexData = {
+        // x  y  z  w
+        0.f, 0.f, 0.f, 1.f,
+    };
+
+    glBufferData(GL_ARRAY_BUFFER, sizeof(VertexData), VertexData.data(), GL_STATIC_DRAW);
+    glVertexAttribPointer(0, 4, GL_FLOAT, GL_FALSE, 4 * sizeof(GLfloat), nullptr);
+    glEnableVertexAttribArray(0);
+    glBindVertexArray(0);
+}
```

Comment:
Would it make sense to just move the contents of the function into the previous function instead?  Since we are only calling it once anyway

Reply:
Yep, sure thing